department-of-veterans-affairs/va.gov-cms
# VACMS-15641 national outreach checkbox

## Description
This PR adds a checkbox to Events content types, that allows a user to publish the Event to the [Outreach calendar](https://www.va.gov/outreach-and-events/events/) (an Event Listing), without having to effect the cardinality of the existing `field_listing` field (a change we previously attempted, but that broke several things unexpectedly).

Relates to:
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10089
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15641

## Testing done & Screenshots
[Tested by Jill](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15641#issuecomment-1773419429).

New Checkbox
![Screenshot 2023-10-23 at 9 54 38 AM](https://github.com/department-of-veterans-affairs/content-build/assets/221539/cc1604d3-6718-4392-81f9-0517350e5080)

## QA steps

As an administrator: check the existence of Feature Toggle
1. Visit /admin/config/system/feature_toggle
   - [x] Validate that the feature toggle `FEATURE_EVENT_OUTREACH_CHECKBOX` is present.
2. Then if the feature is not enabled, check the box next to it and click Save.

As an administrator: check that the checkbox appears, and adds the event to the Outreach calendar.
1. Visit /node/add/event
   - [x] Below the 'Where should the event be listed?' field, verify there is a new 'Publish to the National Outreach Calendar' checkbox.
2. Check the box, and continue filling out the Event.
3. Save the Event.
   - [x] On the View page, under 'Additional Listings', verify you see the 'Outreach Events' link (which links to the /outreach-and-events/events page.

As Winfield.DanielsonIII@va.gov:  check that the checkbox appears for specified users, and adds to the Outreach calendar.
1. Visit /node/add/event
   - [x] Below the 'Where should the event be listed?' field, verify there is a new 'Publish to the National Outreach Calendar' checkbox.
2. Check the box, and continue filling out the Event.
3. Save the Event.
   - [x] On the View page, under 'Additional Listings', verify you see the 'Outreach Events' link (which links to the /outreach-and-events/events page.

As a content-publisher/admin only user (eg: Timothy.Chavis@va.gov) : check that the checkbox DOES NOT appear
1. Visit /node/add/event
   - [x] Below the 'Where should the event be listed?' field, verify there is no 'Publish to the National Outreach Calendar' checkbox.
3. Fill out and Save the Event.
   - [x] On the View page, under 'Additional Listings', verify you do not see the 'Outreach Events' link (which links to the /outreach-and-events/events page.

## Acceptance criteria
- [x] QA steps pass
- [x] ACs from Issue are met
